examples/cpp/helloworld/greeter_server.cc

```diff
@@ -47,19 +59,55 @@ class GreeterServiceImpl final : public Greeter::Service {
     return Status::OK;
   }
 };
+*/
 
 void RunServer() {
   std::string server_address("0.0.0.0:50051");
-  GreeterServiceImpl service;
 
   grpc::EnableDefaultHealthCheckService(true);
   grpc::reflection::InitProtoReflectionServerBuilderPlugin();
+
   ServerBuilder builder;
+
   // Listen on the given address without any authentication mechanism.
   builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+
   // Register "service" as the instance through which we'll communicate with
   // clients. In this case it corresponds to an *synchronous* service.
-  builder.RegisterService(&service);
+
+  // GreeterServiceImpl service;
+  // builder.RegisterService(&service);
+
+  const std::string& path = "other_service_shared.dll"; 
+  // const std::string& path = "C:\\Users\\keithl\\repos\\grpc\\examples\\cpp\\helloworld\\cmake\\build\\Debug\\other_service_shared.dll"; 
+
+  std::wstring temp = std::wstring(path.begin(), path.end());
+  LPCWSTR str = temp.c_str();
+  HINSTANCE hDLL = LoadLibraryW(str);
+
+  if (!hDLL) {
+    throw std::exception("Unable to load DLL");
+  }
+
+  // OtherServiceImpl otherService;
+  // CreateOtherService createOtherService = reinterpret_cast<CreateOtherService>(::GetProcAddress(hDLL, "CreateService"));
+  // CreateOtherService createOtherService = (CreateOtherService)(GetProcAddress(hDLL, "CreateOtherServiceHelper"));
+  // CreateOtherService createOtherService = (CreateOtherService)(GetProcAddress(hDLL, "CreateOtherServiceHelperWithArgs"));
+  SetOtherServicePtr createOtherService = (SetOtherServicePtr)(GetProcAddress(hDLL, "CreateOtherServiceHelperWithArgs"));
+
+  if (!createOtherService) {
+    throw std::exception("Unable to get proc from DLL");
+  }
+
+  // grpc::Service* otherService = createOtherService();
+  // createOtherService();
+
+  grpc::Service* otherService = nullptr;
+  createOtherService(&otherService);
+  
+
+  builder.RegisterService(otherService);
```

Comment:
This is the point in time where we are taking the instance and registering it - if you were to put a breakpoint here, you will see the instance is alive and well, but it seems after first request hits the server, something has gone awry and this service is not funcitonal